cpp/ps0007/problem_s_8.cpp

```diff
@@ -39,17 +42,22 @@ int main() {
         paramCount = 1;
     } else {
         cout << "Shape not recognized." << endl;
-        return 0;
+        return 1;
     }
-
-    unsigned params[paramCount];
+    
+    // fill in the dimensions array 2/3
+    float params[paramCount];
     for (int i = 0; i < paramCount; ++i) {
         cout << "Enter parameter " << i + 1 << ": ";
         cin >> params[i];
+        if (params[i] <= 0)
+            i--;
     }
 
+    // printing out the area of the shape 1/1
     float area = calculateArea(shape, params);
     cout << "The area of the " << shape << " is: " << area << endl;
```

Comment:
- [x] invoked function to calculate the area of the shape
- [x] used `cout::std` in a right way to print out the area